### ISSUE TITLE
#![deny(warnings)] causes misleading error messages

### ISSUE THREAD

**Issue by SimonSapin** [rust-lang/rust#26170]
```rust
#![deny(warnings)]
fn main() {
    (0f64).is_positive();
}
```
In rustc 1.2.0-nightly (d6c8028ce 2015-06-09):
```rust
a.rs:3:12: 3:25 error: use of deprecated item: renamed to is_sign_positive, #[deny(deprecated)] on by default
a.rs:3     (0f64).is_positive();
                  ^~~~~~~~~~~~~
error: aborting due to previous error
```

The error message says "`#[deny(deprecated)]` on by default", which is not true. Instead it should say "enabled here…" and point to line 1.

**Comment by birkenfeld** [comment]
This is fixed in beta and nightly:
```
<anon>:4:12: 4:23 error: use of deprecated item: renamed to is_sign_positive
<anon>:4     (0f64).is_positive();
                    ^~~~~~~~~~~
<anon>:2:8: 2:16 note: lint level defined here
<anon>:2 #[deny(warnings)]
                ^~~~~~~~
```

**Comment by SimonSapin** [comment]
:+1:
